homeassistant/components/ai_speaker/sensor.py

```diff
@@ -0,0 +1,77 @@
+"""Support for AI-Speaker sensors."""
+from datetime import timedelta
+import logging
+
+from aisapi.ws import AisWebService
+
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(minutes=60)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Perform the setup for AI-Speaker status sensor."""
+    _LOGGER.debug("AI-Speaker sensor, async_setup_entry")
+    async_add_entities([AisSensor(hass, config_entry.data)], True)
+
+
+async def async_unload_entry(hass, entry):
+    """Clean up when integrations are removed - add code if you need to do more."""
+    _LOGGER.debug("AI-Speaker sensor, async_unload_entry")
+
+
+class AisSensor(Entity):
+    """AiSpeakerSensor representation."""
+
+    def __init__(self, hass, config_entry_data):
+        """Sensor initialization."""
+        self._ais_info = config_entry_data.get("ais_info")
+        self._ais_id = self._ais_info.get("ais_id")
+        self._ais_url = self._ais_info.get("ais_url")
+        self._web_session = aiohttp_client.async_get_clientsession(hass)
+        self._ais_gate = AisWebService(hass.loop, self._web_session, self._ais_url)
+
+    @property
+    def device_info(self):
+        """Device info."""
+        return {
+            "identifiers": {(DOMAIN, self._ais_id)},
+            "name": "AI-Speaker " + self._ais_info["Product"],
+            "manufacturer": self._ais_info["Manufacturer"],
+            "model": self._ais_info["Model"],
+            "sw_version": self._ais_info["OsVersion"],
+            "via_device": None,
+        }
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, friendly identifier for this entity."""
+        return self._ais_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "AI-Speaker " + self._ais_info["Product"] + " status"
+
+    @property
+    def state(self):
+        """Return the status of the sensor."""
+        return self._ais_info["ApiLevel"]
```

Comment:
What kind of measurement is this?

Reply:
custom one  "api level" added in 
5619c95

Reply:
What is the api level?

I'm not familiar with the AI Speaker. Is there a link to a product on the web?

Reply:
> What is the api level?

The api level is information about supported features, it can be diferent on each device and it can change after the device upgrade or device software update (done by user). 

The goal of this first sensor is to provide the info about the speaker api level - supported features on the speaker. 
This speaker is not only media player but IoT gate and conversation agent on Android platform. We have a plan to integrate this more with Home Assitant. Next step will be media player with media browser, then more sensors, then tts, then conversation agent. So be prepared for long run with me ;) 

> I'm not familiar with the AI Speaker. Is there a link to a product on the web?

We are still in dev phase. Currently we are selling the audio and iot gate which is a target board for AI-Speaker. 
The gate will be incuded into nice lookin wooden speaker - this process is in progress. I hope soon "will be loud" about AI-Speaker ;) 
You can read about us and our current products on our project page - sorry it is in Polish only:
https://ai-speaker.com/

The codes are here - the same licence as HA:
https://github.com/sviete

BTW
To use the AI-Speaker integration in HA (in the basic version) people don't need to buy our device. The free app for Android will be enough.

Reply:
Thanks!